sys/kern/vfs_syscalls.c

```diff
@@ -237,6 +237,18 @@ int do_getdents(proc_t *p, int fd, uio_t *uio) {
   return error;
 }
 
+static int vfs_check_remove(vnode_t *dvp, vnode_t *vp, cred_t *cred) {
+  vattr_t dva;
+  accmode_t diracc = VWRITE;
+  int error;
+  VOP_GETATTR(dvp, &dva);
```

Comment:
You can consider checking if `VOP_GETATTR` fails.

Reply:
I forgot about that. Thanks